shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -693,6 +710,23 @@ private void updateSemantics(ByteBuffer buffer, String[] strings) {
         }
     }
 
+    // Called by native to notify first Flutter frame rendered.
+    private void onFirstFrame() {
+        if (mFirstFrameListeners.isEmpty()) {
```

Comment:
Don't access mFirstFrameListeners on the GPU thread.  Move this to the runnable that is invoked on the platform thread.